src/python/grpcio_tests/tests/unit/_auth_context_test.py

```diff
@@ -0,0 +1,157 @@
+        self._channel = grpc.insecure_channel('localhost:%d' % port)
+        response = self._channel.unary_unary(_UNARY_UNARY)(_REQUEST)
+
+        auth_data = pickle.loads(response)
+        self.assertIsNone(None, auth_data[_ID])
+        self.assertIsNone(None, auth_data[_ID_KEY])
+        self.assertEqual({}, auth_data[_AUTH_CTX])
+
+    def testSecureNoCert(self):
+        self._server_pool = logging_pool.pool(test_constants.THREAD_CONCURRENCY)
+        handler = grpc.method_handlers_generic_handler('test', {
+            'UnaryUnary':
+            grpc.unary_unary_rpc_method_handler(handle_unary_unary)
+        })
+        self._server = grpc.server(
+            self._server_pool,
+            handlers=(handler,),
+            maximum_concurrent_rpcs=test_constants.THREAD_CONCURRENCY)
+        port = self._server.add_secure_port(
+            '[::]:0',
+            grpc.ssl_server_credentials(
+                [(resources.private_key(), resources.certificate_chain())]))
+        self._server.start()
+        self._channel = grpc.secure_channel(
+            'localhost:{}'.format(port),
+            grpc.ssl_channel_credentials(resources.test_root_certificates()), (
+                ('grpc.ssl_target_name_override', _SERVER_HOST_OVERRIDE,),))
+        response = self._channel.unary_unary(_UNARY_UNARY)(_REQUEST)
+
+        auth_data = pickle.loads(response)
+        self.assertIsNone(None, auth_data[_ID])
+        self.assertIsNone(None, auth_data[_ID_KEY])
+        self.assertEqual({'transport_security_type': [b'ssl']},
+                         auth_data[_AUTH_CTX])
+
+    def testSecureClientCert(self):
+        self._server_pool = logging_pool.pool(test_constants.THREAD_CONCURRENCY)
+        handler = grpc.method_handlers_generic_handler('test', {
+            'UnaryUnary':
+            grpc.unary_unary_rpc_method_handler(handle_unary_unary)
+        })
+        self._server = grpc.server(
+            self._server_pool,
+            handlers=(handler,),
+            maximum_concurrent_rpcs=test_constants.THREAD_CONCURRENCY)
+        port = self._server.add_secure_port(
+            '[::]:0',
+            grpc.ssl_server_credentials(
+                [(resources.private_key(), resources.certificate_chain())],
+                root_certificates=resources.test_root_certificates(),
+                require_client_auth=True))
+        self._server.start()
+        self._channel = grpc.secure_channel(
+            'localhost:{}'.format(port),
+            grpc.ssl_channel_credentials(
+                root_certificates=resources.test_root_certificates(),
+                private_key=resources.private_key(),
+                certificate_chain=resources.certificate_chain()), (
+                    ('grpc.ssl_target_name_override', _SERVER_HOST_OVERRIDE,),))
+        response = self._channel.unary_unary(_UNARY_UNARY)(_REQUEST)
+
+        auth_data = pickle.loads(response)
+        self.assertEqual(sorted(_CLIENT_IDS), sorted(auth_data[_ID]))
```

Comment:
The `sorted`s get in the way of seeing what this test is really about. Is `six.assertCountEqual` the right assertion method for this?